src/lib_modules.ml

```diff
@@ -50,9 +50,18 @@ let make_wrapped ~(lib : Dune_file.Library.t) ~dir ~transition ~modules
     else
       wrap_modules modules, Module.Name.Map.empty
   in
+  let implements = Dune_file.Library.is_impl lib in
   let alias_module =
     let alias_prefix =
-      String.uncapitalize (Module.Name.to_string main_module_name) in
+      let base =
+        String.uncapitalize (Module.Name.to_string main_module_name) in
+      if implements then
+        sprintf "%s_%s__" base (Lib_name.Local.to_string (snd lib.name))
```

Comment:
Should we use `"%s__%s__`? To avoid clashes between implementation `bar` of library `foo` and library `foo_bar`

Reply:
Yeah, that makes sense. Although I'm wondering if this will cause issues if we introduce more than 1 layer of aliasing. For example, `Foo.Bar.` We can fix that later however.